src/api/dataops/filteringCondition.ts

```diff
@@ -49,47 +47,73 @@ export class FilteringCondition<U> implements ICondition {
   }
 
   public compile() {
-    return { field: this.field, operator: this.operator, values: this.values, type: this.Type };
+    return [
+      { field: this.field },
+      this.operator,
+      this.value,
+    ];
   }
 }
 
 /**
  * @internal
  */
 export class CompositeFilteringCondition implements ICondition {
-  private logicalOperatorType: string;
+  private logicalOperatorType: LogicalOperator;
   private conditions: ICondition[];
 
-  constructor(filteringCondition: ICondition, logicalOperatorType: string) {
-    this.conditions = [];
-    this.conditions.push(filteringCondition);
+  constructor(filteringCondition: ICondition, logicalOperatorType: LogicalOperator) {
+    this.conditions = [filteringCondition];
     this.logicalOperatorType = logicalOperatorType;
   }
 
-  public get Type(): string{
+  public get type(): LogicalOperator {
     return this.logicalOperatorType;
   }
 
-  public set Type(type: string) {
+  public set type(type: LogicalOperator) {
     this.logicalOperatorType = type;
   }
 
   public or(condition: ICondition): CompositeFilteringCondition {
-    condition.Type = "or";
-    this.conditions.push(condition);
-    return this;
+    return this.appendCondition(condition, "or");
   }
 
   public and(condition: ICondition): CompositeFilteringCondition {
-    condition.Type = "and";
-    this.conditions.push(condition);
-    return this;
+    return this.appendCondition(condition, "and");
   }
 
   public compile() {
-    return {
-      conditions: this.conditions.map( (condition: ICondition) => condition.compile()),
-      type: this.logicalOperatorType,
-     };
+    return this.conditions.reduce(this.toCompiledConditions, []);
+  }
+
+  private toCompiledConditions(expressions: any[], condition: ICondition) {
+    const appendNestedConditions = () => {
```

Comment:
I didn't get the purpose of this function. What is going to change if you open it?

Reply:
The reason of this function it's just to separate the logic of this method (`toCompiledConditions`)

Reply:
Well, I do not insist, but it looks a bit simpler without that separation ;)

Reply:
yeah, I agree it looks simpler, fixed.